compiler/astutil/astutil.go

```diff
@@ -226,3 +233,66 @@ func TakeAddress(info *types.Info, e ast.Expr) *ast.UnaryExpr {
 	SetType(info, ptrType, addrOf)
 	return addrOf
 }
+
+// MakeTypedConstant takes an untyped constant value and makes an AST expression
+// representing it with a concrete type that can represent the constant precisely.
+func MakeTypedConstant(info *types.Info, val constant.Value) ast.Expr {
```

Comment:
I wonder if "TypedConstant" is the right term here. I think I understand why you chose that name--it's returning a typed constant expression. However, if I understand, it functions more as an untyped constant, or at least, it's not synonymous with the Go spec's concept of a typed constant.

I don't have a concrete suggestion for a better name, though.  Just concerned that this might confuse someone reading it in the future.

Reply:
I am also, unfortunately, unable to come up with a better name. The reason I chose this name is because this function is used to convert legit untyped constants into a concrete type that can represent them. So in the general sense the name is correct, although it doesn't always (technically) return a constant or a fully valid AST expression...

Renaming functions is easy though, so if someone thinks of a better name at a later date, I'm more than open to it:)